core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -664,117 +659,82 @@ class KafkaController(val config: KafkaConfig,
   }
 
   /**
-    * This is called in case we need to override/revert an ongoing reassignment.
-    * Note that due to the way we compute the original replica set, we have no guarantee that a revert would put it in the same order.
-    * @return An option of the original replicas prior to the ongoing reassignment. None if there is no ongoing reassignment
-    */
-  private def maybeRevertOngoingReassignment(topicPartition: TopicPartition, reassignedPartitionContext: ReassignedPartitionsContext): Option[Seq[Int]] = {
-    reassignedPartitionContext.ongoingReassignmentOpt match {
-      case Some(ongoingAssignment) =>
-        val originalAssignment = ongoingAssignment.previousAssignment
-        assert(ongoingAssignment.isBeingReassigned)
-        assert(!originalAssignment.isBeingReassigned)
+   * Update the current assignment state in Zookeeper and in memory. If a reassignment is already in
+   * progress, then the new reassignment will supplant it and some replicas will be shutdown.
+   *
+   * Note that due to the way we compute the original replica set, we have no guarantee that a revert
+   * would put it in the same order.
```

Comment:
Let me try to clarify this in the comment since I was also confused by this. I think the problem is specifically when the target replica set overlaps with the existing replica set. For example, suppose you have an initial assignment of [1, 2, 3] and you want to reassign to [3, 4, 2]. While the reassignment is in progress, the target replicas are listed first, so we would have [3, 4, 2, 1]. We can guarantee the desired order when the reassignment completes, but we have lost the order of the original assignment. There's not really an obvious solution to this problem that I can see without adding additional state.

Reply:
+1 to that clarification, this was my thinking when I wrote that comment